src/nupic/algorithms/Connections.hpp

```diff
@@ -41,15 +41,15 @@ namespace nupic
 
     namespace connections
     {
-      typedef UInt32 CellIdx;
-      typedef unsigned char SegmentIdx;
-      typedef unsigned char SynapseIdx;
+      typedef UInt CellIdx;
```

Comment:
With the default TM initializes, 65536 cells are required. But that throws the assert check against CELL_MAX (65536-1). The key change here will only be to CELL_MAX, Segment and Synapse indices can remain with the 8 bit storage. I'll change back all but CELL_MAX, and determine a appropriate value for it.